homeassistant/components/sensor/fitbit.py

```diff
@@ -414,9 +418,16 @@ def unit_of_measurement(self):
     @property
     def icon(self):
         """Icon to use in the frontend, if any."""
-        if self.resource_type == 'devices/battery':
-            return 'mdi:battery-50'
-        return 'mdi:walk'
+        if self.resource_type == 'devices/battery' and self.extra:
+            if self.extra.get('battery') == 'High':
```

Comment:
You could make another dict that maps the strings for battery level to integer levels, then use the util function `icon_for_battery_level` in util.py.